glusterd2/brick/types.go

```diff
@@ -88,13 +107,13 @@ func (b *Brickinfo) Validate(check InitChecks) error {
 		return fmt.Errorf("Brick path %s is not a directory", b.Path)
 	}
 
-	if check.IsMount {
+	if !check.AllowMount {
```

Comment:
I'm finding it slightly hard to reason about these negative if statements with the new names. I prefer `check.IsMount`

The flags exposed to user need not map to internal variable names as is.

Reply:
the request will contain the flags set to true if they want to skip particular steps, for readability I have used negatively if statements.

Reply:
@prashanthpai  addressed review comments. please review